components/camel-jpa/src/main/java/org/apache/camel/component/jpa/JpaConsumer.java

```diff
@@ -97,64 +97,68 @@ protected int poll() throws Exception {
-                    } else {
-                        cause = new PersistenceException(e);
+                    PersistenceException cause = null;
+                    int messagePolled = 0;
+                    try {
+                        messagePolled = processBatch(CastUtils.cast(answer));
+                    } catch (Exception e) {
+                        if (e instanceof PersistenceException) {
+                            cause = (PersistenceException) e;
+                        } else {
+                            cause = new PersistenceException(e);
+                        }
                     }
-                }
 
-                if (cause != null) {
-                    if (!isTransacted()) {
-                        LOG.warn("Error processing last message due: {}. Will commit all previous successful processed message, and ignore this last failure.", cause.getMessage(), cause);
-                    } else {
-                        // Potentially EntityManager could be in an inconsistent state after transaction rollback,
-                        // so disposing it to have it recreated in next poll. cf. Java Persistence API 3.3.2 Transaction Rollback
-                        LOG.info("Disposing EntityManager {} on {} due to coming transaction rollback", entityManager, this);
-                        entityManager.close();
-                        entityManager = null;
-                        
-                        // rollback all by throwning exception
-                        throw cause;
+                    if (cause != null) {
+                        if (!isTransacted()) {
+                            LOG.warn("Error processing last message due: {}. Will commit all previous successful processed message, and ignore this last failure.", cause.getMessage(), cause);
+                        } else {
+                            // rollback all by throwning exception
+                            throw cause;
+                        }
                     }
-                }
 
-                // commit
-                LOG.debug("Flushing EntityManager");
-                entityManager.flush();
-                // must clear after flush
-                entityManager.clear();
-                return messagePolled;
-            }
-        });
+                    // commit
+                    LOG.debug("Flushing EntityManager");
+                    entityManager.flush();
+                    // must clear after flush
+                    entityManager.clear();
+                    return messagePolled;
+                }
+            });
+        } catch (Exception e) {
+            // Potentially EntityManager could be in an inconsistent state after transaction rollback,
+            // so disposing it to have it recreated in next poll. cf. Java Persistence API 3.3.2 Transaction Rollback
+            LOG.info("Disposing EntityManager {} on {} due to coming transaction rollback", entityManager, this);
```

Comment:
I'm ok with that.  It might be seen more often than it was in it's original position and normal behavior doesn't have any particular need to be called out.